File: Otimizados/peraaitalidandoerrado.c
#include <stdio.h>
#include <math.h>

int main(void) {
    while (1) {
        unsigned long long int N;
        scanf("%llu", &N);
        getchar();

        if (N == 0) break;
 
        long double raiz = sqrt(N);
        unsigned long long int raiz_ = (unsigned long long int)raiz;

        if (raiz_ * raiz_ != N) printf("PERA AI... TA LIDANDO ERRADO!\n");
        else {
            unsigned long long int AN = 1 + (raiz - 1) * 2;
            printf("%llu\n", AN);
        }
    }
    return 0;
}

File: peraaitalidandoerrado.py
while True:
    N = int(input())
    if N == 0: break

    raiz = N ** 0.5

    if str(raiz).split('.')[1] != '0': print("PERA AI... TA LIDANDO ERRADO!")
    else:
        AN = 1 + (raiz - 1) * 2
        print(int(AN))
